src/iidm/BusBreakerVoltageLevel.cpp

```diff
@@ -359,6 +359,65 @@ void BusBreakerVoltageLevel::removeTopology() {
     removeAllBuses();
 }
 
+void BusBreakerVoltageLevel::traverse(BusTerminal& terminal, VoltageLevel::TopologyTraverser& traverser) const {
+    std::vector<std::reference_wrapper<Terminal>> traversedTerminals;
+    traverse(terminal, traverser, traversedTerminals);
+}
+
+void BusBreakerVoltageLevel::traverse(BusTerminal& terminal, VoltageLevel::TopologyTraverser& traverser, std::vector<std::reference_wrapper<Terminal>>& traversedTerminals) const {
+    auto it = std::find_if(traversedTerminals.begin(), traversedTerminals.end(), [&terminal](const std::reference_wrapper<Terminal>& t) {
+        return stdcxx::areSame(terminal, t.get());
+    });
+
+    if (it != traversedTerminals.end()) {
+        return;
+    }
+
+    std::vector<std::reference_wrapper<Terminal>> nextTerminals;
+
+    // check if we are allowed to traverse the terminal itself
+    if (traverser.traverse(terminal, terminal.isConnected())) {
+        traversedTerminals.emplace_back(terminal);
+
+        addNextTerminals(terminal, nextTerminals);
+
+        // then check we can traverse terminal connected to same bus
+        unsigned long v = *getVertex(terminal.getConnectableBusId(), true);
+        ConfiguredBus& bus = m_graph.getVertexObject(v);
+
+        const stdcxx::range<BusTerminal>& busTerminals = bus.getTerminals();
+        for (Terminal& t : busTerminals) {
+            if (traverser.traverse(t, t.isConnected())) {
```

Comment:
To avoid infinite loop, we should ensure that `t` is not the same terminal as `terminal` (BusBreakerVoltageLevel.java, line 818)